modules/lambdas/main.tf

```diff
@@ -552,6 +550,73 @@ resource "aws_lambda_function" "orca_catalog_reporting_dummy" {
   }
 }
 
+# API Gateway
+resource "aws_api_gateway_rest_api" "orca_catalog_reporting_api" {
+  name = "${var.prefix}_orca_catalog_reporting_api"
+}
+
+resource "aws_api_gateway_resource" "orca_catalog_reporting_api_resource" {
+  path_part   = "{proxy+}"
+  parent_id   = aws_api_gateway_rest_api.orca_catalog_reporting_api.root_resource_id
+  rest_api_id = aws_api_gateway_rest_api.orca_catalog_reporting_api.id
+}
+
+resource "aws_api_gateway_method" "orca_catalog_reporting_api_method" {
+  rest_api_id   = aws_api_gateway_rest_api.orca_catalog_reporting_api.id
+  resource_id   = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method   = "ANY"
+  # todo: Make sure this is locked down against external access.
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "orca_catalog_reporting_api_integration" {
+  rest_api_id             = aws_api_gateway_rest_api.orca_catalog_reporting_api.id
+  resource_id             = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method             = aws_api_gateway_method.orca_catalog_reporting_api_method.http_method
+  integration_http_method = "POST"
+  type                    = "AWS"
+  uri                     = aws_lambda_function.orca_catalog_reporting_dummy.invoke_arn
+}
+
+resource "aws_api_gateway_method_response" "orca_catalog_reporting_response_200" {
+  rest_api_id = aws_api_gateway_rest_api.orca_catalog_reporting_api.id
+  resource_id = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method = aws_api_gateway_method.orca_catalog_reporting_api_method.http_method
+  status_code = "200"
+}
+
+resource "aws_api_gateway_integration_response" "orca_catalog_reporting_api_response" {
+  depends_on = [aws_api_gateway_integration.orca_catalog_reporting_api_integration]
+  rest_api_id = aws_api_gateway_rest_api.orca_catalog_reporting_api.id
+  resource_id = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method = aws_api_gateway_method.orca_catalog_reporting_api_method.http_method
+  status_code = aws_api_gateway_method_response.orca_catalog_reporting_response_200.status_code
+
+  # Transforms the backend JSON response to XML
+  response_templates = {
+    "application/json" = <<EOF
+#set($inputRoot = $input.path('$'))
+$input.json("$")
+
+#if($input.path("stackTrace") != '')
+#set($context.responseOverride.status = 500)
+#elseif($input.path("httpStatus") != '')
+#set($context.responseOverride.status = $input.path('httpStatus'))
+#end
+EOF
+  }
+}
+
+resource "aws_lambda_permission" "orca_catalog_reporting_api_permission" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.orca_catalog_reporting_dummy.function_name
```

Comment:
Maybe move the parameters under resource to ##OPTIONAL and #REQUIRED sections. See lambda resources for example.

Reply:
I would honestly prefer it if anybody looking to remove/modify these parameters looked at the docs... Since optional params have defaults, I don't think grouping them saves anything.

Reply:
I guess the one time it helps is to identify variables that can definitely NOT be removed? I don't like that it implies that the other variables can be.

Reply:
ya I think moving the parameters under those two sections will help others to know which one cannot be removed.

Reply:
But if you remove any of these, the code won't function as intended... You'll either get a blatant error, or the default value will break something. I'm having trouble thinking of a use case where somebody looks at the tf, sees that some variables are "optional" vs "required", and that informs a decision.